contractcourt/commit_sweep_resolver.go

```diff
@@ -252,18 +252,86 @@ func (c *commitSweepResolver) Resolve(_ bool) (ContractResolver, error) {
 
 		signDesc = c.commitResolution.SelfOutputSignDesc
 	)
+
+	// commitScriptTimeLockTaproot is a function that returns the timelock
+	// of a non-htlc commitment output.
+	//
+	// TODO(proofofkeags): move this
+	// NOTE FOR REVIEWERS: This function does not really belong here. That
+	// said, I don't know where it should go. If you have opinions, please
+	// comment.
+	commitScriptTimeLockTaproot := func(script []byte) fn.Option[int32] {
+		tok := txscript.MakeScriptTokenizer(0, script)
+		tok.Next() // PUSH <key>
+		if len(tok.Data()) != 32 {
+			return fn.None[int32]()
+		}
+		tok.Next() // CHECKSIG
+		if tok.Opcode() != txscript.OP_CHECKSIG {
+			return fn.None[int32]()
+		}
+
+		tok.Next() // PUSH <timelock>
+		var timelock int32
+		switch {
```

Comment:
I think we can break this into two cases:
  1. One where `IsSmallInt` returns true with the argument of `tok.Opcode()`. 
  2. One where that is false. 

Though with the above comment, I think we can get away with _just_ pattern matching on the data push before the checksig.

Reply:
See the main comment above, resolving this.